sdk/core/src/policies/custom_headers_injector_policy.rs

```diff
@@ -0,0 +1,47 @@
+use crate::policies::{Policy, PolicyResult};
+use crate::{PipelineContext, Request, Response};
+use http::header::HeaderMap;
+use std::sync::Arc;
+
+#[derive(Debug, Clone, PartialEq, Eq, Default)]
+pub struct CustomHeaders(HeaderMap);
```

Comment:
Nit: do we also want to make the internal `HeaderMap` pub so folks can just write `CustomHeaders(my_header_map)`?

Reply:
Good idea. 👍